### PR TITLE
packetbeat: nfs: refactor code base for cleaner design

### PR DESCRIPTION
**Pull request by kofemann**
Cleanup some packet processing flow. Create event only when we process a
request. The event population concentrated in a single place. Added handling
of requests without reply. Split request and reply processing into dedicated
methods.

The produced event is not modified.

Signed-off-by: Tigran Mkrtchyan <tigran.mkrtchyan@desy.de>

### CODE REVIEW

**Comment by kofemann** [comment]
my apologies for creating noise. But I want to have a clean code.

File: packetbeat/protos/nfs/request_handler.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	event := common.MapStr{}
+	event["@timestamp"] = common.Time(ts)
+	event["status"] = common.OK_STATUS // all packes are OK for now
+	event["src"] = &src
+	event["dst"] = &dst
+
+	nfsVers := xdr.getUInt()
+	nfsProc := xdr.getUInt()
+
+	// build event only if it's a nfs packet
```
Comment (urso) [inline packetbeat/protos/nfs/request_handler.go]: this comment is a little irritating.

**Comment by urso** [comment]
LGTM. NFS requires quite some more cleanup, but it's a start.

**Comment by kofemann** [comment]
Please feel free to rise your concerns. I am happy to address them. There are another three commits on top of this. But I prefer to post them one-by-one as it's easier to review.

File: packetbeat/protos/nfs/request_handler.go (hunk 1 above)
Comment (kofemann) [inline packetbeat/protos/nfs/request_handler.go]: agree, removed.

**Comment by kofemann** [comment]
@urso @ruflin any update on this ?

**Comment by kofemann** [comment]
see PR(#1670)

**Pull request closed without merging**